### ISSUE TITLE
setUserId on AWSLexInteractionKitConfig does not send userId on Lex request

### ISSUE THREAD

**Issue by saraht129** [awslabs/aws-sdk-ios-samples#218]
Hi there,

I did the following to set up `AWSLexInteractionKit`:

        self.credentialProvider = AWSCognitoCredentialsProvider(regionType: .USEast1, identityPoolId: "test")
        self.configuration = AWSServiceConfiguration(region: .USEast1, credentialsProvider: credentialProvider)
        AWSServiceManager.default().defaultServiceConfiguration = self.configuration

        let chatConfig = AWSLexInteractionKitConfig.defaultInteractionKitConfig(withBotName: "mybot", botAlias: "Dev")
        chatConfig.autoPlayback = false
        chatConfig.userId = token
        AWSLexInteractionKit.register(with: self.configuration!, interactionKitConfiguration: chatConfig, forKey: LexManager.interactionKitKey)

but on my lambda function, the request is as follows:
```
received: LexRequest(currentIntent=Intent(name=QuerySpendingSimple, slots={duration=null, expenseItem=booze, period=null, periodEnd=null}, confirmationStatus=None), userId=us-east-1:b3d44f10-c80c-4dbc-82f7-651801e7896b, inputTranscript=Hello, invocationSource=DialogCodeHook, outputDialogMode=Text, messageVersion=1.0, sessionAttributes=
{}
)
```

with the `userId` value always being the default one, instead of the value that I set.

What is wrong here?

**Comment by sweeneyapps** [comment]
@saraht129   I dig into source code and discovered that you don't have to set userId to chatConfig.   Because credentialsProvider will provide userId to the config automatically.

**Comment by saraht129** [comment]
Thanks. Is there a way on the SDK I can set a custom userId to be sent in `LexRequest`?

**Comment by sweeneyapps** [comment]
your welcome,, I'm looking into this thru source code..  When I find a way.  I'll let you know.

**Comment by sweeneyapps** [comment]
@saraht129 

let key = "userId"
let userId = ""
chatConfig.setGlobalSessionAttribute(userId, forKey: key)

**Comment by sweeneyapps** [comment]
@saraht129 

the below code is from objective-c source code.   So I'm thinking, you might be writing chatConfig.userId correctly.     

maybe you are seeing userId as something..  it's not a token.    It's userId from this example:

```Swift
let credentialProvider = AWSCognitoCredentialsProvider(regionType: .USEast1, identityPoolId: IDENTITY_POOL_ID)
credentialProvider.getIdentityId().continueWith(block: { task in
    print("IdentityId: \(String(describing: task.result))")
    chatConfig.userId = task.result
})
```

i ran test on this, and userId will look like this:  us-east-1:f3c97e7d-2930.....

**Comment by saraht129** [comment]
Yes, I believe the userId you just retrieved is what I'm seeing on `LexRequest` too, but what I want to do is to set `chatConfig.userId` to some other value, for example, the userId of our own system.

**Comment by sweeneyapps** [comment]
I don't know what you are using for token.  Can you try to use String for userId?  

example:   chatConfig.userId = "TestUser"

**Comment by saraht129** [comment]
I've found it. Setting the userId through interaction kit is the way.

     self.interactionKit = AWSLexInteractionKit(forKey: LexManager.interactionKitKey)
     self.interactionKit?.interactionKitConfig.userId = "testId"

**Comment by sweeneyapps** [comment]
@saraht129 that's great.

 Just recently, I ran the sample and test this code:
```chatConfig.setGlobalSessionAttribute("testUserId1234", forKey: "userId")```

and it works well.   (the previous test was when I was running the code without lambda function, that's why I thought it wouldn't work)
